### DISCUSSION TITLE
Recommendation: Rename the `basic/` test folder to `concrete/`

### DISCUSSION THREAD

**Discussion by mds1** [sablier-labs/evm-monorepo#604] [General]
The `basic/` subfolder in test folders refers to what I've (thanks to dapptools) always called "concrete" tests. Concrete tests are your standard deterministic unit test that take no inputs and run the exact same test every time (in contrast to fuzz/property/invariant tests which are not deterministic).

The term concrete is commonly used in the context of symbolic execution. With symbolic execution program inputs are represented as generic symbols to test all execution paths, whereas with concrete execution inputs are fixed and test a specific path. You may have heard of [concolic testing](https://en.wikipedia.org/wiki/Concolic_testing), where the name "concolic" is derived from **conc**rete and symb**olic**

"Concrete" feels like a clear, unambiguous term to refer to deterministic unit tests, whereas "basic" is more vague. I'd suggest renaming "basic" to "concrete" everywhere both to make the architecture in this repo more clear, and to help the ecosystem standardize around a useful terminology

I know you've already gone through some audits and are likely deploying to production soon, so totally understand if you want to avoid a big rename since the codebase is likely largely frozen at this point. But still wanted to bring this up as a broader discussion and something to consider going forward.

**Comment by PaulRBerg** [comment]
Thanks for taking the time to post this discussion and share your insights, @mds1!

> Concrete tests are your standard deterministic unit test that take no inputs and run the exact same test every time

I wish we had known about this term when we wrote the tests! It seems like the perfect fit for what are out tests actually do.

> The term concrete is commonly used in the context of symbolic execution
> ...
> You may have heard of [concolic testing](https://en.wikipedia.org/wiki/Concolic_testing)

TIL

> "Concrete" feels like a clear, unambiguous term to refer to deterministic unit tests, whereas "basic" is more vague

Agree - I've opened an issue for tracking: https://github.com/sablier-labs/v2-core/issues/613

> I know you've already gone through some audits and are likely deploying to production soon, so totally understand if you want to avoid a big rename since the codebase is likely largely frozen at this point

Only `src` is frozen. We were expecting to make changes in the tests.
